src/lib/utils/styling/styler.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {Inject, Injectable, Optional, PLATFORM_ID} from '@angular/core';
+import {isPlatformBrowser, isPlatformServer} from '@angular/common';
+
+import {applyCssPrefixes} from '../auto-prefixer';
+import {ServerStylesheet} from './server-stylesheet';
+import {SERVER_TOKEN} from './server-token';
+
+@Injectable()
+export class StyleService {
```

Comment:
Class description?

On naming: my rule is to avoid the term "Service" in class names because it doesn't tell you anything about what the class actually does. You'll notice in material we have things like `ScrollDispatcher`, `FocusMonitor`, `InteractivityChecker`, etc. instead of `ScrollService`, `FocusService`, `InteracitivityService`. Adding a more specific noun makes the name much more meaningful.

Reply:
I vaguely recall @ThomasBurleson saying the same. The original name was `StyleUtils`, I'll change it back.